whisper/whisperv6/whisper.go

```diff
@@ -176,14 +181,41 @@ func (w *Whisper) SetMaxMessageSize(size uint32) error {
 }
 
 // SetMinimumPoW sets the minimal PoW required by this node
-func (w *Whisper) SetMinimumPoW(val float64) error {
+func (w *Whisper) SetMinimumPoW(val float64, testMode bool) error {
 	if val <= 0.0 {
 		return fmt.Errorf("invalid PoW: %f", val)
 	}
-	w.settings.Store(minPowIdx, val)
+
+	w.notifyPeersAboutPowRequirementChange(val)
+
+	if testMode {
+		w.settings.Store(minPowIdx, val)
+	} else {
+		go func() {
+			//	// allow some time before all the peers have processed the notification
+			time.Sleep(time.Duration(w.reactionAllowance) * time.Second)
+			w.settings.Store(minPowIdx, val)
+		}()
+	}
+
 	return nil
 }
 
+func (w *Whisper) notifyPeersAboutPowRequirementChange(pow float64) {
+	w.peerMu.Lock()
```

Comment:
Doing network stuff in a critical section, that's a long waiting time while holdig a mutex. Could you do something lighter, e.g. starting a goroutine to do this? Or copy the list ? How contentious do we expect that list to be, and how much of a problem is it going to be if a peer is removed/added?